### PR TITLE
Add Status of Findings when Close Finding

### PR DESCRIPTION
**Pull request by italvi**
This is a try to resolve the issue discussed in #7059 

Result:
grafik

### CODE REVIEW

**Comment by italvi** [comment]
@alles-klar as far as I know you are an expert to the k8 deployment: Do you know why it fails?

**Comment by mtesauro** [comment]
I re-ran the failed ones and it seems to be working now.

File: dojo/api_v2/views.py [hunk 1]
```diff
@@ -559,6 +559,9 @@ def close(self, request, pk=None):
                     finding.mitigated = timezone.now()
                 finding.mitigated_by = request.user
                 finding.active = False
+                finding.false_p = finding_close.validated_data['false_p']
+                finding.duplicate = finding_close.validated_data['duplicate']
+                finding.out_of_scope = finding_close.validated_data['out_of_scope']
```
Comment (Maffooch) [inline dojo/api_v2/views.py:564]:
```suggestion
                finding.false_p = finding_close.validated_data.get('false_p, False)
                finding.duplicate = finding_close.validated_data.get('duplicate', False)
                finding.out_of_scope = finding_close.validated_data.get('out_of_scope', False)
```

**Comment by italvi** [comment]
@Maffooch you are right with the .get. Was a bit lazy from my side to do it in one file but not another. I implemented your suggestion and added a missing apostrophe.

**Comment by italvi** [comment]
I have 2 improvements that I would like to add in the future:

1. For the duplicates field there should be the possiblity to choose from findings which is the "original"
2. Remove the possibility to edit the status in the EditForm, s.t. nobody can just change the status without adding a note.

What do you think? Would this be also good PR?

**Review by Maffooch — APPROVED** [review]

**Comment by Maffooch** [comment]
> For the duplicates field there should be the possiblity to choose from findings which is the "original"

This seems sensical to me when manually setting the flag.

> Remove the possibility to edit the status in the EditForm, s.t. nobody can just change the status without adding a note.

I think this would be a little annoying tbh. Especially for smaller teams or folks just getting to know dojo. I think it would be best to leave that to the users

**Comment by italvi** [comment]
> > Remove the possibility to edit the status in the EditForm, s.t. nobody can just change the status without adding a note.
> 
> I think this would be a little annoying tbh. Especially for smaller teams or folks just getting to know dojo. I think it would be best to leave that to the users

@Maffooch could you explain me https://github.com/DefectDojo/django-DefectDojo/blob/master/dojo/finding/views.py#L744 then? It seems to me that somebody had the inital thought of checking whether the finding includes any notes? If so, then the contributor mad a mistake with `closing_disabled!=0`, which should be `closing_disabled==0 ` as notes will be 0 when there are no notes and therefore closing_disabled will remain 0

**Comment by Maffooch** [comment]
Looks like something to do with note types from this #1539 
From just a brief skim of the PR description, it looks like if you have a specific note type configured, it requires that a note of that type be present in the finding  before it is actually closed. I like this approach

**Comment by italvi** [comment]
@mtesauro could you please review the PR too or is something missing?

**Comment by mtesauro** [comment]
@italvi LGTM - approving and merging

**Review by mtesauro — APPROVED** [review]
Approve

**Pull request merged**